Gcd_nobis-julian/src/gcd_nobis/julian/Gcd_nobisJulian.java

```diff
@@ -0,0 +1,105 @@
+                System.out.println("No negative numbers allowed! Sorry!");
+            }
+            else{
+                System.out.println("EUCLEDIAN ALGORITHM:\ngcd of " + number1 + " and " + number2 + " is: " + result);
+            }
+            result = gcdEucledian(number3, number4); // negative example
+            if (result == -1){ // invalid number
+                System.out.println("No negative numbers allowed! Sorry!");
+            }
+            else{
+                System.out.println("EUCLEDIAN ALGORITHM:\ngcd of " + number1 + " and " + number2 + " is: " + result);
+            }
+        }
+        else{ // prime-factors algorithm
+            if (result == -1){ // invalid number
+                System.out.println("No negative numbers allowed! Sorry!");
+            }
+            else{
+                System.out.println("\nPRIME-FACTORS ALGORITHM:\ngcd of " + number1 + " and " + number2 + " is: " + result);
+            }
+            result = gcdPrimeFactors(number3, number4); // negative example
+            if (result == -1){ // invalid number
+                System.out.println("No negative numbers allowed! Sorry!");
+            }
+            else{
+                System.out.println("PRIME-FACTORS ALGORITHM:\ngcd of " + number1 + " and " + number2 + " is: " + result);
+            }
+        }
+    }
+    
+    public static int gcdPrimeFactors(int a, int b){
+        if (a <= 0|| b <= 0) return -1; // not valid
+        // section for 'a' variable
+        int[] remainders_A = new int[10];
+        int temp_A = a-1;
+        int i_A = 0;
+        while (a != 1){ // find out remainders for A
+            while ((a % temp_A) != 0) temp_A--;
+            remainders_A[i_A] = a / temp_A--;
+            a /= remainders_A[i_A++];
+        }
+        // section for 'b' variable
+        int[] remainders_B = new int[10];
+        int temp_B = b-1;
+        int i_B = 0;
+        while (b != 1){ // find out remainders for B
+            while ((b % temp_B) != 0) temp_B--;
+            remainders_B[i_B] = b / temp_B--;
+            b /= remainders_B[i_B++];
+        }   
+        
+        int result = multiplyRemainders(remainders_A, remainders_B);
+        return result; 
+    }
+    // multiply common remainders of BOTH arrayA and arrayB
+    public static int multiplyRemainders(int[]array_A, int[]array_B){
+        int result = 1;
+        for (int i = 0; i < array_A.length && array_A[i] != 0; i++){ // array is partially filled with 0s, therefore check for 0
+            for (int n = 0; n < array_B.length && array_B[n] != 0; n++){ // array is partially filled with 0s, therefore check for 0
+                if (array_A[i] == array_B[n++]){
+                    result *= array_A[i++]; // "primfaktorenzerlegung multiplizieren"
+                }
+            }
+        }
+        return result;
+    }
+    
+    public static int gcdEucledian(int a, int b){
+       if (a == 0 || b == 0) return a+b;

```

Comment:
Here it is not necessary to check for both parameters to be 0. If you check only for b, then you maybe go one step deeper in the recursion but then you will end anyway.